docs/architecture/adr/ADR042-static-ingress-ip-workaround.md

```diff
@@ -0,0 +1,77 @@
+# ADR042: Static ingress IP workaround
+
+## Status
+
+Accepted
+
+## Context
+
+Some production systems have existing requirements for static ingress IPs and
+have external users/companies relying on being able to safelist ingress IPs.
+
+The current ingress architecture is that each namespace within the cluster has
+it's own dynamically provisioned NLB (Network Load Balancer) that directs traffic to
+the namespace's own ingressgateway. [See
+ADR037](./ADR037-per-namespace-gateways.md) for more details on this
+architecture.
+
+Static IP addresses create complications for dynamic, ephemeral infrastructure
+systems such as this and as such we would like to continue to push back on
+these kinds of static configurations that add little to security at a great
+cost to complexity. However in the meantime we need a way to allocation static
+ingress IPs.
+
+We have already provisioned an advertised 5x potential ingress IPs to the
+relevant external parties. But this EIPs have not yet been allocated.
+
+It is not possible to attach EIPs to an existing NLB without recreating it.
+
+Some research in this space:
+
+* [Merged into Kubernetes v1.16.0](https://github.com/kubernetes/kubernetes/pull/69263/commits/7767535426b29fc14461083528b0d15493a3262e)
```

Comment:
👍 links are gr8

Reply:
although... I am skeptical that this will work, given we currently specify `service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags` which has been present since 1.7.0, but isn't actually working for us right now.

Reply:
tag allocation is working ok ... we just have a permission issue viewing tags in console :roll_eyes:

Reply:
Turns out I was wrong, this annotation does work, I just wasn't allowed to see the tags.  Raised https://github.com/alphagov/gsp/pull/758 to fix that.